### ISSUE TITLE
List: move these things upstream to PyTorch, Funsor, etc.

### ISSUE THREAD

**Issue by fritzo** [pyro-ppl/pyro#2630]
This issue tracks Pyro components that should be moved to upstream dependencies.

## Move to PyTorch
- [ ] `dist.Delta` #1781
- [ ] `constraints.IndependentConstraint`
  - usage in `MultivariateNormal`
  - usage in `dist.Independent`
- [ ] `constraints.integer`
- [ ] `Geometric.log_prob()` as patched in pyro/distributions/torch.py
- [ ] `Uniform`'s `._unbroadcasted_*` logic

## Move to Funsor
See also https://github.com/pyro-ppl/funsor/issues/362

- [ ] `distributions.util.detach()`
- [ ] `ops.tensor_utils.haar_transform()` and `.inverse_haar_transform()`

**Comment by RylanSchaeffer** [comment]
@fritzo out of curiosity is there a tutorial on the process for migrating code, updating docs and writing tests?

**Comment by fritzo** [comment]
@RylanSchaeffer there is no tutorial on migrating distributions. You'd need to
- add code to torch/distributions/dirichlet_multinomial.py or torch/distributions/conjugate.py
- fix docstrings to follow numpy syntax rather than sphinx syntax
- import distributions in torch/distributions/__init__.py
- add tests to test/distributions/test_distributions.py
- add an entry in docs/source/distributions.rst
- cc @neerajprad and I on the pull request.

See for example #48798
